crates/ruff/src/rules/pyupgrade/rules/use_pep695_type_alias.rs

```diff
@@ -75,15 +80,78 @@ pub(crate) fn non_pep695_type_alias(checker: &mut Checker, stmt: &StmtAnnAssign)
     //              as type params instead
     let mut diagnostic = Diagnostic::new(NonPEP695TypeAlias { name: name.clone() }, stmt.range());
     if checker.patch(diagnostic.kind.rule()) {
+        let mut visitor = TypeVarNameVisitor {
+            names: vec![],
+            semantic: checker.semantic(),
+        };
+        visitor.visit_expr(value);
+
+        let type_params = if visitor.names.is_empty() {
+            None
+        } else {
+            Some(ruff_python_ast::TypeParams {
+                range: TextRange::default(),
+                type_params: visitor
+                    .names
+                    .iter()
+                    .map(|name| {
+                        TypeParam::TypeVar(TypeParamTypeVar {
+                            range: TextRange::default(),
+                            name: Identifier::new(name.id.clone(), TextRange::default()),
+                            bound: None,
+                        })
+                    })
+                    .collect(),
+            })
+        };
+
         diagnostic.set_fix(Fix::automatic(Edit::range_replacement(
             checker.generator().stmt(&Stmt::from(StmtTypeAlias {
                 range: TextRange::default(),
                 name: target.clone(),
-                type_params: None,
+                type_params,
                 value: value.clone(),
             })),
             stmt.range(),
         )));
     }
     checker.diagnostics.push(diagnostic);
 }
+
+struct TypeVarNameVisitor<'a> {
+    names: Vec<&'a ExprName>,
+    semantic: &'a SemanticModel<'a>,
```

Comment:
Nit: we typically call this argument `model`.

Reply:
A search for `semantic.` gives 303 results while `model.` gives 28 results / `semantic: &Semantic` gives 99 results while `model: &Semantic` gives 13. You've told me this before but it doesn't seem to be the case. Happy to consolidate on a standard though.

Reply:
I prefer semantic or semantic_model. Model is too generic in my view

Reply:
I must be mixing up the convention, feel free to move forward with `semantic: &Semantic`, my mistake.

Reply:
https://github.com/astral-sh/ruff/pull/6406